Sources/Commands/UserToolchain.swift

```diff
@@ -110,45 +121,14 @@ public struct UserToolchain: Toolchain {
             throw Error.invalidToolchain(problem: "could not find `clang` at expected path \(clangCompiler.asString)")
         }
 
-        // Find the default SDK (on macOS only).
-      #if os(macOS)
-        let sdk: AbsolutePath
-
-        if let value = lookupExecutablePath(filename: getenv("SYSROOT")) {
-            sdk = value
-        } else {
-            // No value in env, so search for it.
-            let foundPath = try Process.checkNonZeroExit(
-                args: "xcrun", "--sdk", "macosx", "--show-sdk-path").chomp()
-            guard !foundPath.isEmpty else {
-                throw Error.invalidToolchain(problem: "could not find default SDK")
-            }
-            sdk = AbsolutePath(foundPath)
-        }
-
-        // Verify that the sdk exists and is a directory
-        guard localFileSystem.exists(sdk) && localFileSystem.isDirectory(sdk) else {
-            throw Error.invalidToolchain(problem: "could not find default SDK at expected path \(sdk.asString)")
-        }
-        defaultSDK = sdk
-
-        // Try to get the platform path.
-        let platformPath = try? Process.checkNonZeroExit(
-            args: "xcrun", "--sdk", "macosx", "--show-sdk-platform-path").chomp()
-        if let platformPath = platformPath, !platformPath.isEmpty {
-            sdkPlatformFrameworksPath = AbsolutePath(platformPath)
-                .appending(components: "Developer", "Library", "Frameworks")
-        } else {
-            sdkPlatformFrameworksPath = nil
-        }
-      #else
-        defaultSDK = nil
-      #endif
+        self.swiftPlatformArgs = UserToolchain.generateSwiftPlatformArgs(destination: destination)
+        self.clangPlatformArgs = UserToolchain.generateClangPlatformArgs(destination: destination)
```

Comment:
We can compute the arguments inline here or keep the computed property, don't need this static method.